src/main/java/io/vertx/core/Vertx.java

```diff
@@ -473,6 +474,24 @@ static void clusteredVertx(VertxOptions options, Handler<AsyncResult<Vertx>> res
   <T> void executeBlocking(Handler<Future<T>> blockingCodeHandler, Handler<AsyncResult<T>> resultHandler);
 
   /**
+   * Add an interceptor that will be called whenever work is scheduled
+   *
+   * @param interceptor  the interceptor
+   * @return a reference to this, so the API can be used fluently
+   */
+  @GenIgnore
+  Vertx addSchedulerInterceptor(BiFunction<Context, Runnable, Runnable> interceptor);
```

Comment:
At some point we should discuss supporting this in a polyglot way (meaning supporting BiFunction in codegen).